packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2765,18 +2768,32 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 return { stage: "generate", ...generateSummaryData, error: continueResult.error };
             }
 
-            const summaryContext: ISummaryContext =
-                lastAck === undefined
-                    ? {
-                        proposalHandle: undefined,
-                        ackHandle: this.context.getLoadedFromVersion()?.id,
-                        referenceSequenceNumber: summaryRefSeqNum,
-                    }
-                    : {
-                        proposalHandle: lastAck.summaryOp.contents.handle,
-                        ackHandle: lastAck.summaryAck.contents.handle,
-                        referenceSequenceNumber: summaryRefSeqNum,
-                    };
+            // It may happen that the lastAck it not correct due to missing summaryAck in case of single commit
+            // summary. So if the previous summarizer closes just after submitting the summary and before
+            // submitting the summaryOp then we can't rely on summaryAck. So in case we have
+            // latestSnapshotVersionId from storage and it does not match with the lastAck ackHandle, then use
+            // the one fetched from storage as parent as that is the latest.
+            let summaryContext: ISummaryContext;
+            if (lastAck?.summaryAck.contents.handle !== latestSnapshotVersionId
+                && latestSnapshotVersionId !== undefined) {
+                summaryContext = {
+                    proposalHandle: undefined,
+                    ackHandle: latestSnapshotVersionId,
+                    referenceSequenceNumber: summaryRefSeqNum,
+                };
+            } else if (lastAck === undefined) {
+                summaryContext = {
+                    proposalHandle: undefined,
+                    ackHandle: this.context.getLoadedFromVersion()?.id,
+                    referenceSequenceNumber: summaryRefSeqNum,
+                };
+            } else {
+                summaryContext = {
+                    proposalHandle: lastAck.summaryOp.contents.handle,
+                    ackHandle: lastAck.summaryAck.contents.handle,
+                    referenceSequenceNumber: summaryRefSeqNum,
+                };
+            }
```

Comment:
Is it the same as: 
            if (lastAck === undefined) {
                summaryContext = {
                    proposalHandle: undefined,
                    ackHandle: latestSnapshotVersionId === undefined ? this.context.getLoadedFromVersion()?.id : latestSnapshotVersionId,
                    referenceSequenceNumber: summaryRefSeqNum,
                };
            } else {
                summaryContext = {
                    proposalHandle: lastAck.summaryOp.contents.handle,
                    ackHandle: lastAck.summaryAck.contents.handle !== latestSnapshotVersionId ? latestSnapshotVersionId : lastAck.summaryAck.contents.handle,
                    referenceSequenceNumber: summaryRefSeqNum,
                };

Reply:
+ 1 on this.
I'd also add that 

> latestSnapshotVersionId === undefined ? this.context.getLoadedFromVersion()?.id : latestSnapshotVersionId

can be replaced with 

> latestSnapshotVersionId ?? this.context.getLoadedFromVersion()?.id

Same for the other case

Reply:
The correct version would be like this:
`			if (lastAck === undefined) {
                summaryContext = {
                    proposalHandle: undefined,
                    ackHandle: latestSnapshotVersionId ?? this.context.getLoadedFromVersion()?.id,
                    referenceSequenceNumber: summaryRefSeqNum,
                };
			} else {
                summaryContext = {
                    proposalHandle: lastAck?.summaryAck.contents.handle !== latestSnapshotVersionId && latestSnapshotVersionId !== undefined:
						undefined : lastAck?.summaryOp.contents.handle,
                    ackHandle: lastAck?.summaryAck.contents.handle !== latestSnapshotVersionId && latestSnapshotVersionId !== undefined:
						latestSnapshotVersionId : lastAck?.summaryAck.contents.handle,
                    referenceSequenceNumber: summaryRefSeqNum,
                };
            }`

This seems very complicated, so I will prefer the current code over it as it clearly tells the 3 cases.

Reply:
But I am fine with both. So let me know if you have any hard inclination towards the other one.